lib/ansible/galaxy/collection.py

```diff
@@ -310,6 +310,14 @@ def from_name(collection, apis, requirement, force, parent=None):
 
         for api in apis:
             collection_url_paths = [api.api_server, 'api', 'v2', 'collections', namespace, name, 'versions']
+
+            available_api_versions = get_available_api_versions(api)
+            if 'v3' in available_api_versions:
+                # /api/v3/ exists, use it
+                collection_url_paths[2] = 'v3'
+                # update this v3 GalaxyAPI to use Bearer token from now on
+                api.token_type = 'Bearer'
```

Comment:
Does it make sense for `Bearer` to be set in `_auth_header()` based on the version used instead of you manually setting it each time for v3 endpoints.